frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,19 @@
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import { useAppSelector, useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+
+type Properties = {
+  children: React.ReactNode;
+};
+const ProtectedRoute: React.FC<Properties> = ({ children }: Properties) => {
+  const { auth } = useAppSelector((users) => users);
```

Comment:
RootState is not `users` and you do not need to destructure the whole RootState - you need only `auth`